### DISCUSSION TITLE
Platform Release v0.27.4

### DISCUSSION THREAD

**Discussion by connorjak** [SimDynamX/SpaceTeamsPRO#64] [Announcements]
### This is copy pasted from the SimDynamX internal repository, so most or all of the links here direct to non-public sites.

This new Space Teams PRO version has been released and is available for download on [spaceteamspro.com](https://www.spaceteamspro.com/).

Space Teams PRO releases that start with `0.` are all _Beta_ releases, meaning we have relaxed standards for stability, performance, and backwards compatibility as we quickly introduce new features. Please contact us directly (support@SimDynamX.com) if you are interested in using Space Teams in an environment where software stability is critical.

## Most notable changes since 0.27.3:
- Far object point-rendering 
  - Planets and some moons will still be rendered as a point when too far away to be reliably drawn as geometry
  - Proper brightness and color depending on location and lighting angle
  - Similar technique to star rendering
  - Will be extended to be available for all other entities in the future
- Added Entity_Camera class
  - All image-export cameras are now entities, and can be positioned and rotated as is usual for Entities.
  - New method for Camera Image Export based on Events
  - C++ `st_camera::CaptureImage()` and `st_camera::OnImageReceived()` functions
  - Sometimes causes crashes when used through Python API; not ready for use in STU Lunar SAR Task 2
- Several fixes for STU Lunar SAR Task 2
- Login screen handles no-cached-credentials more cleanly

## Key Known Issues
- SpaceCRAFT Project Management Git UI doesn't automatically refresh to show results of `Clone`/`Pull`/`Push`
- https://github.com/SimDynamX/SC_Platform/issues/1491 https://github.com/SimDynamX/SC_Platform/issues/2235 Realtime Capture SkyLight sometimes has improper lighting
- Various small GMRAD issues involving cube-side edges https://github.com/SimDynamX/SC_Platform/issues/783 
- https://github.com/SimDynamX/SC_Platform/issues/1739 Some jitter in specific situations, probably involving a time synchronization issue
- Lander demo was mistakenly modified; the version that's in the demo will not control itself
- Planet surface collision not yet reimplemented https://github.com/SimDynamX/SC_Platform/issues/2336
- Create New Sim UI is only partially implemented
- Don't yet have a way to set up Systems on Entities in the SpaceCRAFT sim editing utilities; still requires text-editing the sim config for now.

# Automated Git Changelog:

## What's Changed
* Platform Release v0.27.2 by @connorjak in https://github.com/SimDynamX/SC_Platform/pull/2511
* Orbit Pawn Attach to Entity Settings Gadget by @Laurenbrady01 in https://github.com/SimDynamX/SC_Platform/pull/2509
* Far object point-rendering by @connorjak in https://github.com/SimDynamX/SC_Platform/pull/2517
* Camera Frame Export refactor by @connorjak in https://github.com/SimDynamX/SC_Platform/pull/2523
* Possible NaN fix in BodyCentered_LVLH, other misc changes for FDIR by @connorjak in https://github.com/SimDynamX/SC_Platform/pull/2526

**Full Changelog**: https://github.com/SimDynamX/SC_Platform/compare/v0.27.3...v0.27.4

## Do not download the "Assets" files below; they are just a copy of this repository and are not used for Space Teams Pro. Space Teams Pro can be downloaded after making an account at [spaceteamspro.com](https://www.spaceteamspro.com/).

<hr /><em>This discussion was created from the release <a href='https://github.com/SimDynamX/SpaceTeamsPro/releases/tag/v0.27.4'>Platform Release v0.27.4</a>.</em>

**Comment by rcghpge** [comment]
Hey for this key known issue, 'SpaceCRAFT Project Management Git UI doesn't automatically refresh to show results of Clone/Pull/Push' - are there issues pushing changes to Github repos and fetching the main repos on Github? My Git UI on STP is not updating changes I am trying to push locally to Github

**Reply by connorjak** [reply]
You just need to press Refresh after doing a Pull or a Push. Otherwise, the pull/push will still happen; it will just not show that anything changed yet. May be able to fix for the upcoming camera fix release.

**Reply by rcghpge** [reply]
are there features to delete cloned repos on the STP Git UI? I'm having merging conflicts on Git thru the STP Git UI

**Reply by connorjak** [reply]
Not yet. You can press the Open Local Assets Folder button to open that folder and either use git / github desktop yourself to resolve merge conflicts, or delete individual repo folders manually.

**Reply by rcghpge** [reply]
Hey Connor, There are issues with functionality with VS Code. I am having scripting issues for dependencies such as the spaceteams library or numpy in EntityBehavior_LTV.py where I have to specify the file path of the Python Interpreter. Tinkering away though lol gg

**Reply by connorjak** [reply]
Maybe you haven't done FirstTimeSetup.bat on this new version? Here's the relevant docs: https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-API-Docs#how-to-set-up-your-development-environment

**Comment by rcghpge** [comment]
Are y'all's going to implement a simulation crash feature for STP maybe in future releases?

**Reply by connorjak** [reply]
What do you mean by simulation crash?

**Reply by rcghpge** [reply]
like a traceback error UI feature in STP

**Reply by connorjak** [reply]
We have tracebacks for errors that happen in the python code, which print to that Python System's log file. For all other errors, we have C++ exceptions that get caught and handled by our Logger code and end up in a log file. We don't yet have code that attempts to retrieve log files from other processes and display that, but that is the next step. We will eventually have some time to attempt that.

**Reply by rcghpge** [reply]
That would be awesome. Streamline the workflow in the future 🙏
